google/cloud/spanner_dbapi/connection.py

```diff
@@ -421,6 +455,7 @@ def begin(self):
                 "is already running"
             )
         self._transaction_begin_marked = True
+        self._transaction_isolation_level = isolation_level
```

Comment:
Shouldn't  `_transaction_isolation_level` fallback to `isolation_level` set at the connection if `isolation_level` is not provided in begin method ?

Reply:
Yes, but we do that at a slightly later moment when the transaction is actually being created here: https://github.com/googleapis/python-spanner/blob/82d15ba4fc1b38f4a604267b93a614bf20dfb031/google/cloud/spanner_dbapi/connection.py#L394

The reason for doing it like this is that this allows us to support a statement like `set transaction isolation level repeatable read` even after calling `begin`. (We don't support this yet in dbapi as it is part of the PostgreSQL dialect, but we do support this in for example JDBC).

The use of the default from the connection is covered by this test case: https://github.com/googleapis/python-spanner/blob/82d15ba4fc1b38f4a604267b93a614bf20dfb031/tests/mockserver_tests/test_dbapi_isolation_level.py#L71